rust/emitter/src/ast_emitter.rs

```diff
@@ -202,6 +202,12 @@ impl AstEmitter {
             }
 
             Expression::LiteralStringExpression { value, .. } => {
+                // Hack: Prevent emission for scripts with "use strict"
+                // directive.
+                if value == &"use strict" {
+                    return Err(EmitError::NotImplemented("TODO: use strict directive"));
+                }
+
```

Comment:
this needs to be moved to `AstBuilder`, given `Script` has `directives` field, and `"use strict"` should be put there, instead of `statements` field.
So, if emitter finds "use strict" that can be directive while emitting `statements` field, that's `AstBuilder`'s fault.